### PR TITLE
All api groups

### PR DESCRIPTION
**Pull request by qbalukom**

### CODE REVIEW

**Review by dariadomagala — CHANGES_REQUESTED** [review]
You cannot type anything in the input now, it's automatically added as a nwe api gruop etc

File: core-ui/src/components/Predefined/Create/Roles/helpers.js [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+export const EMPTY_API_GROUP_KEY = 'core-api-group';
+export const extractApiGroup = groupVersion => {
+  // handle core ('') group
+  if (groupVersion === 'v1') return '';
+  const [apiGroup] = groupVersion.split('/');
+  return apiGroup;
+};
+
+export const getApiGroupInputOptions = groupVersions =>
+  unique(groupVersions.map(extractApiGroup)).map(g =>
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Create/Roles/helpers.js:13]: can we please add either `?` before map, or `|| []` at the end? I think it's the most common cause when it comes to crashing UIs in our code 😄 

(please also apply in other places, it doesn't make sense to comment it all - also for 'some', 'filter' etc)

File: core-ui/src/shared/ResourceForm/fields/ComboboxArrayInput.js [hunk 2]
```diff
@@ -41,38 +43,59 @@ export function ComboboxArrayInput({
...
+              _ref={ref}
+              selectedKey={value}
+              typedValue={value || ''}
+              selectionType="manual"
+              setValue={setValue}
+              options={filteredOptions()}
+              onKeyDown={focus}
+              onBlur={onBlur}
+              onSelectionChange={(_, selected) => {
+                console.log('onSelectionChange', selected);
```
Comment (dariadomagala) [inline core-ui/src/shared/ResourceForm/fields/ComboboxArrayInput.js:83]: nope

**Review by dariadomagala — CHANGES_REQUESTED** [review]
Now you can type in some values, but those are not being added to the list, nothing happens. I cannot add a verb that's not on the list and it was a reason why we've chosen this component.

File: core-ui/src/shared/ResourceForm/fields/ComboboxArrayInput.js (hunk 2 above)
Comment (qbalukom) [reply core-ui/src/shared/ResourceForm/fields/ComboboxArrayInput.js:83]: oops

**Review by dariadomagala — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**